pkg/registry/networking/ipaddress/strategy.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipaddress
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/networking"
+	"k8s.io/kubernetes/pkg/apis/networking/validation"
+)
+
+// ipAddressStrategy implements verification logic for Replication.
+type ipAddressStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating Replication IPAddress objects.
+var Strategy = ipAddressStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+// Strategy should implement rest.RESTCreateStrategy
+var _ rest.RESTCreateStrategy = Strategy
+
+// Strategy should implement rest.RESTUpdateStrategy
+var _ rest.RESTUpdateStrategy = Strategy
+
+// NamespaceScoped returns false because all IPAddresses is cluster scoped.
+func (ipAddressStrategy) NamespaceScoped() bool {
+	return false
```

Comment:
This is where i think we need to give it more thoughts. There is a lot of things that can use `IPAddress` either within the system or operator. Are we ok saying that anybody wants to use ipaddress object will have to use use it non-namespaced?

I have no opinion as of now, but i would like us to make an informed decision on that one.

Reply:
I've discussed this in the original KEP and the problem is cross-namespaced referencing, that seems to be not a thing ... or a complex thing at least

Reply:
... also the ClusterIP of a service is unique in a cluster, having IPAddress namespaced and apply the restriction that has to be unique  will be very confusing for users

Reply:
ack.